scipy/stats/stats.py

```diff
@@ -3692,6 +3695,163 @@ def f_oneway(*args, axis=0):
     return F_onewayResult(f, prob)
 
 
+def alexandergovern(*args, nan_policy='propagate'):
+    """
+    Performs the Alexander Govern test.
+
+    The Alexander-Govern approximation tests the equality of k independent
+    means in the face of heterogeneity of variance. The test is applied to
+    samples from two or more groups, possibly with differing sizes.
+
+    Parameters
+    ----------
+    sample1, sample2, ... : array_like
+        The sample measurements for each group.  There must be at least
+        two samples.
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle when input contains nan.
+        The following options are available (default is 'propagate'):
+
+          * 'propagate': returns nan
+          * 'raise': throws an error
+          * 'omit': performs the calculations ignoring nan values
+
+    Returns
+    -------
+    statistic : float
+        The computed A statistic of the test.
+    pvalue : float
+        The associated p-value from the chi-squared distribution.
+
+    Notes
+    -----
+    The use of this test relies on several assumptions.
+
+    1. The samples are independent.
+    2. Each sample is from a normally distributed population.
+    3. Unlike `f_oneway`, this test does not assume on homoscedasticity,
+       instead relaxing the assumption of equal variances.
+
+    Input samples must be finite, one dimensional, and with size greater than
+    one. Inputs that are not one dimensional will be flattened.
```

Comment:
I'm sorry, @swallan; I thought that this behavior was common in `scipy.stats`. It is not unprecedented - there are at least two hypothesis tests without an `axis` argument that currently do this - but other functions without an `axis` argument _do_ produce an error unless `ndim` is 1. Can we go back to that behavior?

This is also important because when an `axis` argument _is_ added to this function, we will want the default `axis` to be `0` to be consistent with the other `scipy.stats` functions that accept an `axis` argument.  If we flatten the arrays now, when `axis` is added the default would have to be `None` (which means flatten the input), and this would be very inconsistent.

Reply:
I see. Easy to undo, and it adds the relevance of the 1d error back in which code is already implemented for.